### ISSUE TITLE
"az monitor app-insights component create" fails with HTTP Error

### ISSUE THREAD

**Issue by ChrisKlug** [Azure/azure-cli-extensions#3011]
## Describe the bug

**Command Name**
`az monitor app-insights component create
Extension Name: application-insights. Version: 0.1.13.`

**Errors:**
```
'HTTPError' object has no attribute '_message'
Traceback (most recent call last):
cliextensions/application-insights/azext_applicationinsights/custom.py, ln 74, in create_or_update_component
    return client.create_or_update(resource_group_name, application, component)
mgmt_applicationinsights/v2020_02_02_preview/operations/_components_operations.py, ln 359, in create_or_update
...
cliextensions/application-insights/azext_applicationinsights/custom.py, ln 76, in create_or_update_component
    ex.error._message = ex.error._message + HELP_MESSAGE
AttributeError: 'HTTPError' object has no attribute '_message'
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az monitor app-insights component create --app {} --location {} --kind {} -g {} --workspace {}`

## Expected Behavior
An Application Insights resource is created and connected to a workspace

## Environment Summary
```
Linux-5.4.72-microsoft-standard-WSL2-x86_64-with-debian-bullseye-sid
Python 3.6.10
Installer: DEB

azure-cli 2.19.1

Extensions:
application-insights 0.1.13

```

**Comment by yonzhan** [comment]
monitor

**Comment by ChrisKlug** [comment]
Trying to first create an Insights resource without a workspace, and then update it using `az monitor app-insights component update --app {} -g {} --workspace "/subscriptions/XXXXXXXXXXXXX/resourcegroups/my-resource-group/providers/microsoft.operationalinsights/workspaces/<my-workspace>"` causes __cli.azure.cli.core.azclierror: Operation failed with status: 'Bad Request'. Details: Could not retrieve Log Analytics workspace.__. Even if the workspace definitely exists

**Comment by kairu-ms** [comment]
> Trying to first create an Insights resource without a workspace, and then update it using `az monitor app-insights component update --app {} -g {} --workspace "/subscriptions/XXXXXXXXXXXXX/resourcegroups/my-resource-group/providers/microsoft.operationalinsights/workspaces/<my-workspace>"` causes **cli.azure.cli.core.azclierror: Operation failed with status: 'Bad Request'. Details: Could not retrieve Log Analytics workspace.**. Even if the workspace definitely exists

Hi ChrisKlug. Please use the name of workspace instead of ID and try again:

```
az monitor app-insights component update --app {} -g {} --workspace "my-workspace"
```

BTW we've already written a [test case](https://github.com/Azure/azure-cli-extensions/blob/9da9f2e22aa1ca8dc0af1e3411808fec1e86ac0d/src/application-insights/azext_applicationinsights/tests/latest/test_applicationinsights_mgmt.py#L300-L341) similar to your case. Maybe it can help you.

**Comment by ChrisKlug** [comment]
Using just the name caused it to not be found, which makes me believe it uses the same resource group. And in my case, they are not in the same group

**Comment by kairu-ms** [comment]
> Using just the name caused it to not be found, which makes me believe it uses the same resource group. And in my case, they are not in the same group

I see. Are them in same location?

**Comment by ChrisKlug** [comment]
They are both West Europe, in the same subscription

**Comment by kairu-ms** [comment]
> West Europe

I've test your scenario (same location and subscription but different resource group), it works for my. 
![image](https://user-images.githubusercontent.com/69238381/108676066-9ee3d000-7522-11eb-8e21-a0204da8acba.png)

**Comment by ChrisKlug** [comment]
Yeah, it seems like it just started working for some reason on Friday. No changes to the scripts or anything. It just "magically" started working again... Doesn't inspire too much confidence though when it stops and starts working intermittently with the same code... But thanks for looking into it!

**Comment by kairu-ms** [comment]
> Yeah, it seems like it just started working for some reason on Friday. No changes to the scripts or anything. It just "magically" started working again... Doesn't inspire too much confidence though when it stops and starts working intermittently with the same code... But thanks for looking into it!

It must be caused by an issue on service. Hi @sameergMS, @dadunl. can you help with this issue.

**Comment by piquet-h** [comment]
Same issue here with a customer.

Different log analytics workspace in a different subscription and resource group per Cloud Adoption Framework. I don't think it's a CLI issue, I think it's a platform issue as it can be replicated in the portal also.

**Comment by PeterButzelaar** [comment]
I had the same problem. First succesfully retrieving the workspace:

```
resource logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2021-06-01' existing = {
  name: 'my-workspace'
  scope: resourceGroup('otherResourceGroup')
}
```

When I used the **logAnalyticsWorkspace.id** property to use in the Application Insights resource, it gave me the error:

Could not retrieve Log Analytics workspace.

Turned out, when I output the id to the console, Bicep or Azure CLI put an additional } at the end of the resource:

```
      "workspaceId": {
        "type": "String",
        "value": "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/shared-dev-rg/providers/Microsoft.OperationalInsights/workspaces/my-workspace}"
      }
```

So to solve this, I now have:

`WorkspaceResourceId: replace(logAnalyticsWorkspace.id, '}', '')`

Hope this helps someone.

**Comment by iharmaiseyeu** [comment]
Faced the same error during deployment of template spec that creates Application Insights.

We have 2 subscriptions: 

Subscription 1 contains template spec
Subscription 2 uses the template spec from the Subscription 1 for deployment.

It is not reproducible from portal or when you deploy template spec to the same subscription.

**Comment by merijng** [comment]
@ChrisKlug @kairu-ms Old topic, any new insights lately?

Looks like an underlying issue, within the Azure platform. I have the same problem, but with [Bicep templates](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/overview?tabs=bicep). It fails the first time, as far as I know a second BICEP-deployment does it's job. This might explain why it "magically" started working again (comment feb 22, 2021).

**Bicep setup:**

`
resource logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2020-08-01' = {
  name: 'loganalytics-nameofproject'
  location: location
  properties: {
    sku: {
      name: 'PerGB2018'
    }
    retentionInDays: 120
    features: {
      searchVersion: 1
      legacy: 0
      enableLogAccessUsingOnlyResourcePermissions: true
    }
  }
}

resource appInsights 'Microsoft.Insights/components@2020-02-02' = { 
  kind: 'web'
  location: location
  name: 'appInsights-nameofproject'
  properties: {
    Application_Type: 'web'
    WorkspaceResourceId: '/subscriptions/${subscription().subscriptionId}/resourceGroups/${resourceGroup().name}/providers/Microsoft.OperationalInsights/workspaces/loganalytics-nameofproject'
  }  
}

`

Can it be the same issue?

**Comment by rahulsaxena1987** [comment]
I am also seeing same issue (intermittent) as @merijng . I am simply deploying a bicep template consisting of a log analytics workspace and application insight and deployment fails occasionally saying 'Could not find the log analytics workspace' though the workspace has been deployed fine already.

@ChrisKlug , @kairu-ms - Do you have any recent update on this issue please. Thanks !!

**Comment by merijng** [comment]
@AllyW and @kairu-ms and CC @rahulsaxena1987

I would not focus too much on this specific error. Now I had a different one, for a new project. It fails the first time, and the second time it works.
```Cannot find Subscription with name {subscriptionid}```

It is still on the **Update insights component** activity, in the **Activity log** of Azure. Bicep template is still as described in my previous post.

hopefully this helps :)

**Comment by luisgmgouveia** [comment]
I have the same issue and it didn't work even on the 2nd try.

I'm running this from an azure pipeline. The service connection I use provides me access to the Resource Group A but not to Resource Group B where the Log Analytics Workspace lies. Is this a problem?

I don't want to create a new Log Analytics Workspace. I just want to deploy an Application Insights in RG A that references the Log Analytics Workspace in RG B. 

Do I need to update my service connection to provide access not only to RG A but also to RG B?

Thanks!
